src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  demage(amout) {
+    this.health -= amout;
+
+    if (this.health <= 0) {
+      const indexOfThis = Animal.alive.indexOf(this);
+
+      Animal.alive.splice(indexOfThis, 1);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] - The method `demage` uses `indexOf` and `splice` to remove the animal from the `Animal.alive` array when its health is less than or equal to 0. This can be inefficient due to the use of nested loops. Consider using the `filter` method to create a new array excluding the dead animal.